packages/api/src/event/queries/get_checklist_events.sql

```diff
@@ -0,0 +1,106 @@
+WITH account_events AS (
+  SELECT event.*
+  FROM
+    event
+  INNER JOIN
+    account
+    ON
+      account.subject::uuid = event.actor_id
+  WHERE
+    account.primaryemail = :email
+)
+
+SELECT
+  EXISTS(
+    SELECT 1
+    FROM
+      account_events
+    WHERE
+      account_events.actor_type = 'user'
+      AND account_events.entity = 'archive'
+      AND account_events.action = 'create'
+  ) AS "archiveCreated",
+  EXISTS(
+    SELECT 1
+    FROM
+      account_events
+    WHERE
+      account_events.actor_type = 'user'
+      AND account_events.entity = 'account'
+      AND account_events.action = 'submit_promo'
+      AND account_events.body ->> 'promoCode' = 'TellYourStory'
+  ) AS "storageRedeemed",
+  EXISTS(
+    SELECT 1
+    FROM
+      account_events
+    WHERE
+      account_events.actor_type = 'user'
+      AND account_events.entity = 'legacy_contact'
+      AND account_events.action = 'create'
+  ) AS "legacyContact",
+  EXISTS(
+    SELECT 1
+    FROM
+      account_events
+    WHERE
+      account_events.actor_type = 'user'
+      AND account_events.entity = 'directive'
+      AND account_events.action = 'create'
+  ) AS "archiveSteward",
+  EXISTS(
+    SELECT 1
+    FROM
+      account_events
+    WHERE
+      account_events.actor_type = 'user'
+      AND account_events.entity = 'profile_item'
+      AND account_events.action = 'update'
+  ) AS "archiveProfile",
+  EXISTS(
+    SELECT 1
+    FROM
+      account_events
+    WHERE
+      account_events.actor_type = 'user'
+      AND account_events.entity = 'record'
+      AND account_events.action = 'submit'
+  ) AS "firstUpload",
+  EXISTS(
+    SELECT 1
+    FROM
+      account_events
+    WHERE
+      account_events.actor_type = 'user'
+      AND (
+        (
+          account_events.entity = 'record'
+          AND account_events.action = 'submit'
+          AND (account_events.body ->> 'public')::boolean = true
+        )
+        OR
+        (
+          account_events.entity = 'record'
+          AND account_events.action = 'move'
+          AND (account_events.body ->> 'public')::boolean = true
+        )
+        OR
+        (
+          account_events.entity = 'record'
+          AND account_events.action = 'copy'
+          AND (account_events.body ->> 'public')::boolean = true
+        )
+        OR
+        (
+          account_events.entity = 'folder'
+          AND account_events.action = 'move'
+          AND (account_events.body ->> 'public')::boolean = true
+        )
+        OR
+        (
+          account_events.entity = 'folder'
+          AND account_events.action = 'copy'
+          AND (account_events.body ->> 'public')::boolean = true
+        )
+      )
+  ) AS "publishContent";
```

Comment:
This is nice! What happens if they create an empty folder in Public? (I don't know if that appears as an empty folder in their public archive anyway.)

Reply:
An empty folder would appear in a Public archive, so arguably this should include a check for the creation of public folders. On the other hand, maybe it should be excluding moves and copies of empty folders. @k8lyn6, should adding an empty folder to the public workspace count as publishing?

Reply:
@liam-lloyd hmmmm. I don't think it should, but I also think it's an edge case. Let's go with no.